src/main/java/com/connexta/poller/service/Poller.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.poller.service;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class Poller {
+
+  private final StatusService statusService;
+  private final BlockingQueue<URI> transferQueue;
+  private ExecutorService executor;
```

Comment:
Inject the ExecutorService instead of instantiating it in this class

Reply:
Poller class removed. Executor injected into StatusServiceImpl component.